File: Arrays/One Dimensional Arrays/maxSubArray.py
arr = [1, 8, -7, 22, -15]

l = len(arr)

currSum = 0
maxSum = 0

left = -1
right = -1
for i in range(l):
    for j in range(i, l):
        currSum = 0
        for k in range(i , j + 1):
            currSum += arr[k]
        if (currSum > maxSum):
            maxSum = currSum
            left = i
            right = j

for k in range(left, right + 1):
    print(arr[k], end=',')

print()
print(maxSum)

File: Arrays/One Dimensional Arrays/maxSubArray.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n;
    cin >> n;


    int arr[5] = {1, 7, -4,-3, 8};

    int currSum = 0;
    int maxSum = 0;

    int left = -1;
    int right = -1;

    for (int i = 0; i < n; i++){
        for (int j = i; j < n; j++){
            currSum = 0;
            for (int k = i; k <= j; k++){
                currSum += arr[k];
                
            }
            if (currSum > maxSum){
                maxSum = currSum;
                left = i;
                right = j;
            }
        }
    }

    cout << maxSum << endl;

    for (int k = left ; k <= right; k++ ){
        cout << arr[k] << " , ";
    }
    return 0;
}
